pkg/gardenadm/botanist/machines.go

```diff
@@ -48,14 +48,14 @@ var addressTypePreference = map[corev1.NodeAddressType]int{
 	corev1.NodeHostName: 1,
 }
 
-// PreferredAddressForMachine returns the preferred address of the given machine based on addressTypePreference.
+// PreferredAddress returns the preferred address of the given machine based on addressTypePreference.
 // If the machine has no addresses, an error is returned.
-func PreferredAddressForMachine(machine *machinev1alpha1.Machine) (string, error) {
-	if len(machine.Status.Addresses) == 0 {
-		return "", fmt.Errorf("no addresses found in status of machine %s", machine.Name)
+func PreferredAddress(addresses []corev1.NodeAddress) (string, error) {
+	if len(addresses) == 0 {
+		return "", fmt.Errorf("no addresses found in status of machine")
```

Comment:
The function description as well as the error message are very specific to machines, but the function is now also used with nodes. Should we adapt it accordingly?